### ISSUE TITLE
(No such file or directory) Failed to connect to tmux session. error: exit status 1

### ISSUE THREAD

**Issue by agolomoodysaada** [brosandilabs/kubectl-tmux-ssh#2]
OS: MacOS HighSierra 10.13.5

When running

```
brew install tmux
git clone git@github.com:brosandilabs/kubectl-tmux-ssh.git ~/.kube/plugins/kubectl-tmux-ssh
kubectl plugin tmux-ssh
```

Output
```
error connecting to /private/tmp/tmux-501/default (No such file or directory)
Failed to connect to tmux session.
error: exit status 1
```

Any ideas what I can do to resolve this?

**Comment by adamglt** [comment]
You're running the command without any sessions running (this plugin doesn't spawn a new session).  
Just enter a session yourself then run the command:
```
$ tmux
$ kubectl plugin tmux-ssh <any other flags>
```

**Comment by agolomoodysaada** [comment]
ok I typed `tmux` and opened a session and then tried

```
$ kubectl plugin tmux-ssh
No matching nodes found!
error: exit status 1
```

while

```
$ kubectl get nodes
NAME                        STATUS    ROLES     AGE       VERSION
node-0    Ready     agent     18d       v1.10.3
node-1    Ready     agent     19d       v1.10.3
node-2    Ready     agent     19d       v1.10.3

```

**Comment by adamglt** [comment]
By the "agent" role I'm guessing you're using Azure, I'm not sure what address nodes use there.

Run this:
`kubectl get nodes -o go-template --template='{{range .items}}{{println .metadata.name "\n---"}}{{range .status.addresses}}{{println .type "->" .address}}{{end}}{{println}}{{end}}'`

By the type of output you see there, you can modify the tmux-ssh command. For example, in my example (private topology) I see output like:
```
ip-xxx-xxx-xxx-xxx.eu-west-1.compute.internal
---
InternalIP -> xxx.xxx.xxx.xxx
InternalDNS -> xxx.xxx.xxx.xxx.eu-west-1.compute.internal
Hostname -> xxx.xxx.xxx.xxx.eu-west-1.compute.internal
...
```

So I use `kubectl plugin tmux-ssh -a 'InternalIP'` (or any of the others if you can resolve those DNS entries locally)

**Comment by agolomoodysaada** [comment]
Yes, I am running on Azure Kubernetes Service (AKS). Here's the output from running 

```
kubectl get nodes -o go-template --template='{{range .items}}{{println .metadata.name "\n---"}}{{range .status.addresses}}{{println .type "->" .address}}{{end}}{{println}}{{end}}'
```
in the format below for each node
```
aks-agentpool-xxxxxxxx-x
---
InternalIP -> xxx.xxx.xxx.xxx
Hostname -> aks-agentpool-xxxxxxxx-x
```

Now I don't have any tunnel of VPN so I can't use the InternalIPs as they are private IPs. The same goes for the hostnames. Is there anyway I can do this via the public IP of the nodes?

**Comment by adamglt** [comment]
How do you SSH to your nodes today? have an public IP for them? do you use a bastion host?  
If the node API object doesn't expose an `ExternalIP`, it means k8s isn't aware of that IP (and so this tool can't see it either).  
If I remember correctly, normally it is passed on node registration.

What this tool supports is one of two (very popular) options:
1. Nodes with public IP, visible to k8s itself through the node's `ExternalIP`
2. Nodes with private IP/hostname, visible to k8s itself through the node's `InternalIP` or `Hostname`, coupled with a VPN into the VPC, or a public bastion host used as an SSH-proxy.

**Comment by agolomoodysaada** [comment]
Can you please document this information in the `README`?
You can close this once the README is updated. Thank you!
